### PR TITLE
Added travis setup, essentially verbatim from RAJA

### PR DESCRIPTION
**Pull request by rhornung67**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+sudo: required
+dist: trusty
+language: cpp
+env:
+  global:
+    - secure: xN+lGOH8LzepE1OoOrHelDgu1wf7nL/B7nBWhN7EnCB7S7hZJL/AakruHy4lMfQfF4XkrnPWmBlmc4wdLH+o6jPkUISm4nLRSTMnRV2L+Mjyzg3aIEua0xpO6rLUNgsShB8mfkieTJq+kSj3Yp2CM7GEzm+UNNxeJcY0VdUHy9msRRRbXiLViIrfwBEVC9He7xG9NWfqkpsORfoiPmVDm7YzuXALdB8qkX4AWggysz/BCVj0PwBMr754eEpOodQ9GeKDF2Kwy5vPAqK5f7zwshJtF9VevyA1A2M9y8BHJMymz4wGaSxLNMeUU85AmVIvmzX0weG94JQ7mlUVszNpO5CCIyjwCOF+IDUI8HCDJGOY7+gGnv4H2LhDwAXvFLD65FlMntQQe2e4KRTnFxtJvvghjv5FyxJSHwctLsgeDpr2uZDcAhK1yf8TNsqqMiXQj2yGLByJy8j5PjUyd8oN47uZo0T5DDMd5c3ztUppc5+DisIoqmoYQeom3lYbpeudaf492ZDBWEV4rS9COl1h7CnpanMBpXWLFc2zXyfTpRn3GifutiF8M3rSS2KHcPyb9JLePTrC4+itMkwB4SHo1VYk4H2RQAdPMDFHMKCeVs2Z4sF9pGPJR+JzRekaKFLDm73ihsuE0cnx1oPVQMjSWa0e7A1a9W4UQBvp9xR++i4=
```
Comment (davidbeckingsale) [inline .travis.yml]: Note that this is the Intel license Jeff Hammond gave us for testing RAJA on Travis....

**Comment by davidbeckingsale** [comment]
@rhornung67 you should modify the repository settings so that merges require passing tests and review approval 😄

File: .travis.yml (hunk 1 above)
Comment (rhornung67) [reply .travis.yml]: Is it OK to use for this?

**Comment by rhornung67** [comment]
@davidbeckingsale I'm still setting things up....  BTW, I made you an admin on this project, so you can make configuration mods too.

File: .travis.yml (hunk 1 above)
Comment (willkill07) [reply .travis.yml]: The license should be good for at least two concurrent builds. I know intel has a hard limit of 5 that you can “manage” through the license center. I’m not sure if this license behaves the same way

Comment (davidbeckingsale) [reply .travis.yml]: I'm assuming it's okay, since this is RAJA. I'm not sure if we are subject to the concurrent constraints @willkill07 mentioned.

**Comment by davidbeckingsale** [comment]
I think the errors you are seeing are related to the use of modules with clang.

**Comment by rhornung67** [comment]
You're very confident :1st_place_medal:

**Comment by rhornung67** [comment]
@davidbeckingsale or @jonesholger The checks have now passed twice in a row for this. Can I get a review and approval so I can merge?  Thanks @davidbeckingsale for the CUB cmake fixes. They seem to have done the trick.

**Comment by rhornung67** [comment]
@ajkunen This is working now. I think David and Holger are out. Can you approve so I can merge and move on to other things?

**Pull request merged**